### PR TITLE
Improve `rake activestorage:install` 

### PR DESCRIPTION
**Pull request by bogdanvlviv**
* Set `:environment` as prerequisite `rake activestorage:install`
* Use ActiveRecord::Migrator.migrations_paths.first instead of "db/migrate"
* <s>If `db/migrate` exists, don't try to mkdir `db/migrate`</s>
* Respect `config.active_record.timestamped_migrations`
* Prevent adding migration more than one time

### CODE REVIEW

File: activestorage/lib/tasks/activestorage.rake [hunk 1]
```diff
@@ -2,12 +2,23 @@ require "fileutils"
...
 namespace :activestorage do
   desc "Copy over the migration needed to the application"
-  task :install do
-    migration_file_path = "db/migrate/#{Time.now.utc.strftime("%Y%m%d%H%M%S")}_active_storage_create_tables.rb"
-    FileUtils.mkdir_p Rails.root.join("db/migrate")
-    FileUtils.cp File.expand_path("../../active_storage/migration.rb", __FILE__), Rails.root.join(migration_file_path)
-    puts "Copied migration to #{migration_file_path}"
+  task install: :environment do
+    destination = Rails.root.join("db/migrate")
+    FileUtils.mkdir_p(destination) unless File.exist?(destination)
```
Comment (georgeclaghorn) [inline activestorage/lib/tasks/activestorage.rake]: Why is the existence check necessary? `FileUtils.mkdir_p` won’t raise an error if the given directory already exists.

File: activestorage/lib/tasks/activestorage.rake [hunk 2]
```diff
@@ -2,12 +2,23 @@ require "fileutils"
...
 
-    puts "Now run rails db:migrate to create the tables for Active Storage"
+    migration_class_name = "ActiveStorageCreateTables"
+    migrations = ActiveRecord::Migrator.migrations(destination)
+
+    if migrations.detect { |migration| migration.name == migration_class_name }
+      raise ActiveRecord::MigrationAlreadyExists.new(migration_class_name)
```
Comment (georgeclaghorn) [inline activestorage/lib/tasks/activestorage.rake]: Why raise an error here instead of printing a more helpful message?

**Comment by dhh** [comment]
Until we have a better approach, we should also run this rake task after the app generator is finished such that new apps get the migration automatically.

File: activestorage/lib/tasks/activestorage.rake (hunk 1 above)
Comment (bogdanvlviv) [inline activestorage/lib/tasks/activestorage.rake]: I agree this check is redundant. I will remove it

**Comment by bogdanvlviv** [comment]
Close in favor of #30111

**Pull request closed without merging**